src/main/java/com/erigitic/jobs/TEAction.java

```diff
@@ -0,0 +1,207 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.erigitic.jobs;
+
+import ninja.leaping.configurate.ConfigurationNode;
+import org.slf4j.Logger;
+import org.spongepowered.api.block.BlockState;
+import org.spongepowered.api.block.trait.BlockTrait;
+
+import java.util.*;
+
+public class TEAction {
+
+    private String action;
+    private String targetId;
+
+    private String growthTrait = null;
+    private String idTrait = null;
+    private Map<String, TEActionReward> rewards = new HashMap<>();
+
+    private TEActionReward reward;
+
+    public void loadConfigNode(String action, ConfigurationNode node) {
+        ConfigurationNode idTraitNode = node.getNode("id-trait");
+        ConfigurationNode growthTraitNode = node.getNode("growth-trait");
+
+        if (idTraitNode.isVirtual()) {
+            this.action = action;
+            this.targetId = node.getKey().toString();
+            this.reward = new TEActionReward();
+            this.reward.loadConfigNode(node);
+            this.growthTrait = growthTraitNode.getString(null);
+            return;
+        }
+
+        String traitName = idTraitNode.getString("type");
+
+        ConfigurationNode idTraits = node.getNode(traitName);
+        rewards.clear();
+
+        if (idTraits.hasMapChildren()) {
+            idTraits.getChildrenMap().forEach((k, v) -> {
+                if (!(k instanceof String)) {
+                    return;
+                }
+
+                TEActionReward reward = new TEActionReward();
+                reward.loadConfigNode(v);
+                rewards.put(((String) k), reward);
+            });
+        }
+        this.action = action;
+        this.targetId = node.getKey().toString();
+        this.idTrait = idTraitNode.getString("type");
+        this.growthTrait = growthTraitNode.getString(null);
+    }
+
+    public boolean isGrowing() {
+        return growthTrait != null;
+    }
+
+    public boolean isIdTraited() {
+        return idTrait != null;
+    }
+
+    public String getIdTrait() {
+        return idTrait;
+    }
+
+    public String getAction() {
+        return action;
+    }
+
+    public String getTargetId() {
+        return targetId;
+    }
+
+    public Optional<TEActionReward> getReward() {
+        return Optional.ofNullable(reward);
+    }
+
+    public Map<String, TEActionReward> getRewards() {
+        return rewards;
+    }
+
+    public boolean isValid() {
+        return action != null
+               && !action.trim().isEmpty()
+               && targetId != null
+               && !targetId.trim().isEmpty();
+    }
+
+    public Optional<TEActionReward> evaluateBreak(Logger logger, BlockState state, UUID blockCreator) {
```

Comment:
Getters and setters should be near the end of the class. Move this function, as well as the other functions below it, above the getters and setters.